join.md

```diff
@@ -99,14 +101,21 @@ Join Semantics
 --------------
 
 ```k
+    syntax Bool ::= isLiveGemJoin(String) [function, functonal]
+ // -------------------------------
+    rule [[ isLiveGemJoin(GEMID) => B ]]
+         <gem-join-gem> GEMID </gem-join-gem>
+         <gem-join-live> B </gem-join-live>
```

Comment:
Since none of the other `live` flags are accessed with a `[[ ... ]]` function, and it's only needed in one place, let's just directly match on the `<gem-join-live>` cell where it's needed.

Also, anytime we are matching on a `Map` cell, I always include the outer cell to make it clearer.

```
<gem-join>
  <gem-join-gem>
  <gem-join-live>
  ...
</gem-join>
```

I'm converting over other parts of the semantics which omit this, please do the same.